Cabal/Distribution/PackageDescription/Parse.hs

```diff
@@ -161,6 +162,16 @@ storeXFieldsPD _ _ = Nothing
 -- ---------------------------------------------------------------------------
 -- The Library type
 
+subLibFieldDescrs :: [FieldDescr Library]
+subLibFieldDescrs =
+  [ simpleField "library"
+                        showToken             parseTokenQ
+                        -- The invariant for printing a 'Library' @lib@ is that
+                        -- @isJust (libName lib)@.
+                        (fromMaybe (error "subLibFieldDescrs") . libName)
+                        (\xs lib -> lib{libName=Just xs})
+  ] ++ libFieldDescrs
+
```

Comment:
Oh yes this got killed in the pretty-printer patch. Let me see if I can swizzle the hunks around.